From 44cd589c8a0f5a245e0003a7d0c4be1b5f3ba890 Mon Sep 17 00:00:00 2001
From: Varun Reddy Yeturu <varunreddy.yeturu@codeaurora.org>
Date: Tue, 26 Sep 2017 15:08:49 -0700
Subject: qcacld-3.0: Check for upper bound in P2P NOA event

Check for the upper bounds for number of NOA descriptors
received in the P2P NOA event.

Change-Id: Id7ecf064f2c25f378f76d795902713da8520507f
CRs-Fixed: 2113072
---
 core/wma/src/wma_power.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/core/wma/src/wma_power.c b/core/wma/src/wma_power.c
index 4a5bc0e..83a5a36 100644
--- a/core/wma/src/wma_power.c
+++ b/core/wma/src/wma_power.c
@@ -1528,6 +1528,11 @@ int wma_p2p_noa_event_handler(void *handle, uint8_t *event,
 		descriptors = WMI_UNIFIED_NOA_ATTR_NUM_DESC_GET(p2p_noa_info);
 		noa_ie.num_descriptors = (uint8_t) descriptors;
 
+		if (noa_ie.num_descriptors > WMA_MAX_NOA_DESCRIPTORS) {
+			WMA_LOGD("Sizing down the no of desc %d to max",
+					noa_ie.num_descriptors);
+			noa_ie.num_descriptors = WMA_MAX_NOA_DESCRIPTORS;
+		}
 		WMA_LOGD("%s: index %u, oppPs %u, ctwindow %u, num_descriptors = %u",
 			 __func__, noa_ie.index,
 			 noa_ie.oppPS, noa_ie.ctwindow, noa_ie.num_descriptors);
-- 
cgit v1.1

